lib/server.js

```diff
@@ -1484,8 +1486,24 @@ function errEvtNameFromError(err) {
  * @returns {Function} factory
  */
 function serverMethodFactory(method) {
+    /**
+     * @private
+     * @param {String|Object} opts - path string or config object
+     * @param {String} [opts.path] - path
+     * @param {String} [opts.url] - alias for path
+     * @returns {String} name of the route
+     * @example
+     * server.get('/example/near/:lat-:lng/radius/:r', (req, res, next) => {});
+     * server.get({ path: '/foo' }, (req, res, next) => {});
+     * server.get({ url: '/foo' }, (req, res, next) => {});
+     */
     return function serverMethod(opts) {
-        if (opts instanceof RegExp || typeof opts === 'string') {
+        // RegExp deprecation message
+        if (opts instanceof RegExp) {
+            new TypeError(REGEXP_PATH_DEPRECATED_TEXT);
```

Comment:
Is it deprecated or completely broken? Did you mean to throw this error?

Reply:
It's completely removed in 7x.

Reply:
Cool, then shouldn't we throw here then?

Reply:
oops, that was the plan :D